ext/bcmath/libbcmath/src/div.c

```diff
@@ -31,219 +31,436 @@
+	 * First, numerator_high, which only exists in the numerator, uses its maximum value.
+	 * Considering carry-back, numerator_high can be expressed as follows.
+	 * numerator_high = divisor_high * B
+	 * Also, numerator_low is only present in the numerator, but since this is a subtraction, use the smallest possible value here, 0.
+	 * numerator_low = 0
+	 *
+	 * MAX_E = (numerator_high * divisor_low - divisor_high * numerator_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *       = (divisor_high * B * divisor_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *
+	 * divisor_low uses the maximum value.
+	 * divisor_low = B^k - 1
+	 * MAX_E = (divisor_high * B * divisor_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *       = (divisor_high * B * (B^k - 1)) / (divisor_high * (divisor_high * B^k + B^k - 1))
+	 *
+	 * divisor_high uses the minimum value, but want to see how the number of digits affects the error, so represent
+	 * the minimum value as:
+	 * divisor_high = 10^x (any x ∈ ℕ)
+	 * Since B = 10^b, the formula becomes:
+	 * MAX_E = (divisor_high * B * (B^k - 1)) / (divisor_high * (divisor_high * B^k + B^k - 1))
+	 *       = (10^x * 10^b * ((10^b)^k - 1)) / (10^x * (10^x * (10^b)^k + (10^b)^k - 1))
+	 *
+	 * Now let's make an approximation. Remove -1 from the numerator. Approximate the numerator to be
+	 * large and the denominator to be small, such that MAX_E is less than this expression.
+	 * Also, the denominator "(10^b)^k - 1" is never a negative value, so delete it.
+	 *
+	 * MAX_E = (10^x * 10^b * ((10^b)^k - 1)) / (10^x * (10^x * (10^b)^k + (10^b)^k - 1))
+	 * MAX_E < (10^x * 10^b * (10^b)^k) / (10^x * 10^x * (10^b)^k)
+	 *       < 10^b / 10^x
+	 *       < 10^(b - x)
+	 *
+	 * Therefore, found that if the number of digits in base B and divisor_high are equal, the error will be less
+	 * than 1 regardless of the number of digits in the value of k.
+	 *
+	 * Here, B is BC_VECTOR_BOUNDARY_NUM, so adjust the number of digits in high part of divisor to be BC_VECTOR_SIZE + 1.
+	 */
+	size_t divisor_top_digits = divisor_len % BC_VECTOR_SIZE;
+	if (divisor_top_digits == 0) {
+		divisor_top_digits = BC_VECTOR_SIZE;
+	}
+	size_t high_part_shift = POW_10_LUT[BC_VECTOR_SIZE - divisor_top_digits + 1];
+	size_t low_part_shift = POW_10_LUT[divisor_top_digits - 1];
+	BC_VECTOR tmp_divisor = divisor_vector[divisor_top_index] * high_part_shift + divisor_vector[divisor_top_index - 1] / low_part_shift;
+	for (size_t i = 0; i < quot_arr_size; i++) {
+		BC_VECTOR tmp_numerator = numerator_vector[numerator_top_index - i] * high_part_shift + numerator_vector[numerator_top_index - i - 1] / low_part_shift;
+
+		/* If it is clear that divisor is greater in this loop, then the quotient is 0. */
+		if (div_carry == 0 && tmp_numerator < tmp_divisor) {
+			quot_vector[quot_top_index - i] = 0;
+			div_carry = numerator_vector[numerator_top_index - i];
+			numerator_vector[numerator_top_index - i] = 0;
+			continue;
+		}
```

Comment:
Still going through the algorithm, but why must the `div_carry` also be 0 here?
```suggestion
		/* tmp_numerator < tmp_divisor => quotient == 0 */
		if (div_carry == 0 && tmp_numerator < tmp_divisor) {
			quot_vector[quot_top_index - i] = 0;
			div_carry = numerator_vector[numerator_top_index - i];
			numerator_vector[numerator_top_index - i] = 0;
			continue;
		}
```

Reply:
Basically, the `div_carry` affects the `tmp_numerator` later?

Reply:
Yes, I fixed it to calculate first

https://github.com/php/php-src/pull/14660/commits/dfb1e4fb0b36a5d05160245b7ee37f73e7639a85
https://github.com/php/php-src/pull/14660/commits/f24a72bb68cf8af7d2add328643749a58b9f201c